### ISSUE TITLE
Sequelize doing LEFT OUTER JOIN for candidate info

### ISSUE THREAD

**Issue by aabmass** [brownac/AmExHackathon#23]
This looks to be what sequlize is doing to join the image URIs. I'm no sql master, but isn't outer join suboptimal @Bduong30? Probably not a big deal, just wondering ;)

``` sql
SELECT `candidateInfo`.`id`, 
       `candidateInfo`.`firstname`, 
       `candidateInfo`.`lastname`, 
       `candidateInfo`.`email`, 
       `candidateInfo`.`phonenumber`, 
       `candidateInfo`.`school`, 
       `candidateInfo`.`major`, 
       `candidateInfo`.`graduationdate`, 
       `candidateInfo`.`needsponsorship`, 
       `candidateInfo`.`internorfull`, 
       `candidateInfo`.`areaofinterest`, 
       `candidateInfo`.`preferredlanguages`, 
       `candidateInfo`.`finalevaluation`, 
       `image_uris`.`id`      AS `image_uris.id`, 
       `image_uris`.`img_uri` AS `image_uris.img_uri`, 
       `image_uris`.`type`    AS `image_uris.type` 
FROM   `candidateinfos` AS `candidateInfo` 
       LEFT OUTER JOIN `image_uris` AS `image_uris` 
                    ON `candidateInfo`.`id` = `image_uris`.`id`; 
```

**Comment by Bduong30** [comment]
@aabmass No, I don't think so. The images table has a foreign key 'id', and also can't really be inserted unless there was a successful candidateInfo insertion, so there's no point in having an outer join. In essence the optimal join should really be an inner join, but at the moment it seems we can still add a candidate even if there isn't a picture with it (don't know if I should make an issue for that, the fields don't get reset but the information is still inserted into the database). Now that I think about it though, I guess it would add the candidate id into the image table with a null uri, so an inner join would still work... 
TL;DR: Left outer join is optimal if we want to add candidates without resume pics, otherwise I think it should be an inner join. Outer join won't work cause we will never have images without their corresponding candidates.

**Comment by aabmass** [comment]
Okay!
